client/extensions/woocommerce/woocommerce-services/views/carrier-accounts/dynamic-settings-form.js

```diff
@@ -84,19 +83,35 @@ export const DynamicCarrierAccountSettingsForm = ( props ) => {
 			setIsSaving(true);
 
 			try {
-				await api.post( siteId, api.url.shippingCarrier(), {
+				const result = await api.post( siteId, api.url.shippingCarrier(), {
+					...formValues,
 					type: carrierType,
-					settings: formValues
 				} );
+
+				if( ! result.success ) {
+					throw new Error();
+				}
+
+				successNotice( translate( 'Your carrier account was connected successfully.' ) )
```

Comment:
Added a success notice + redirect, to make it consistent with the UPS account registration behavior